model/src/main/kotlin/Provenance.kt

```diff
@@ -76,7 +76,7 @@ data class Provenance(
         if (sourceArtifact != null) {
             // If the (non-empty) hashes match, we do not need to compare the other properties like the URL. Only
             // support this for a known algorithm whose hash we were able to verify as otherwise the hash could be fake.
```

Comment:
If we accept unknown algorithms now this comment is not true anymore. Or use `canVerify` below instead of `!= NONE`.

Reply:
I've refactored the code slightly to make this check nicer to implement. I'm still not super happy myself, but I believe this is a good intermediate step for further hash refactoring I plan to do.

Reply:
You could make `verifiable` a constructor argument of the enum which defaults to true, and only set it to false for `UNKNOWN` and `NONE`. Maybe that would be nicer, as the value is stored with the enum objects and not in a separate set?